dashboard/app/helpers/aichat_comprehend_helper.rb

```diff
@@ -8,19 +11,68 @@ def self.create_comprehend_client
 
   # Moderate given text for inappropriate/toxic content using AWS Comprehend client.
   def self.get_toxicity(text, locale)
-    comprehend_response = create_comprehend_client.detect_toxic_content(
-      {
-        text_segments: [{text: text}],
-        language_code: locale,
-      }
-    )
-    categories = comprehend_response.result_list[0].labels
+    text_segment_lists = get_text_segment_lists(text)
+    all_results = []
+    text_segment_lists.each do |list|
+      comprehend_response = create_comprehend_client.detect_toxic_content(
+        {
+          text_segments: list.map {|segment| {text: segment}},
+          language_code: locale,
+        }
+      )
+      all_results.concat(comprehend_response.result_list)
+    end
+
     {
       text: text,
-      toxicity: comprehend_response.result_list[0].toxicity,
-      max_category: categories.max_by(&:score),
+      toxicity: all_results.max_by(&:toxicity).toxicity,
+      max_category: all_results.map(&:labels).flatten.max_by(&:score)
     }
   end
+
+  # Returns a collection of text segment lists, each with a list of individual text segments to check for toxicity.
+  # Comprehend limits the amount of bytes per segment, and the amount of overall bytes per list.
+  def self.get_text_segment_lists(text)
```

Comment:
For whatever reason, my local env not connecting. Would you mind checking this level if you have time (I kept receiving this input error on this level)? Thanks! '/s/gen-ai-staging/lessons/14/levels/2'

Reply:
Got my local env working and checked locally. Could get a nice long recipe for dinner tonight! 🎉